packages/now-cli/test/dev/integration.js

```diff
@@ -1347,3 +1347,15 @@ test(
     await testPath(200, `/api/user.sh`, /Hello, from Bash!/m);
   })
 );
+
+test(
+  '[vercel dev] Should work with nested `tsconfig.json` files',
+  testFixtureStdio(
+    'nested-tsconfig',
+    async testPath => {
+      await testPath(200, `/`, /Nested tsconfig.json test page/);
+      await testPath(200, `/api`, 'Nested `tsconfig.json` API endpoint');
```

Comment:
How does this ensure the `api/tsconfig.json` file was used?

Reply:
Because the root level `tsconfig.json` causes `ts-node` to have syntax errors (I believe due to `module: 'esnext'`). The fact that the API endpoint runs successfully is indicative that the proper tsconfig was used.